portalnet/src/utils/db.rs

```diff
@@ -95,11 +97,29 @@ fn get_default_data_dir() -> anyhow::Result<PathBuf> {
 }
 
 /// Returns application private key.
+/// If the private key is provided via env var, it is used.
 /// If the private key does not exist (eg. brand new trin data dir),
 /// a random pk is generated and stored.
 fn get_application_private_key(trin_data_dir: &Path) -> anyhow::Result<CombinedKey> {
-    let unsafe_private_key_file = trin_data_dir.join(UNSAFE_PRIVATE_KEY_FILE_NAME);
-    if !unsafe_private_key_file.exists() {
+    let unsafe_private_key_file = match env::var(TRIN_PRIVATE_KEY_ENV_VAR) {
```

Comment:
I'm still not sure this works. The whole idea behind the `trin_data_dir` is that it's tied directly to the private key identity contained within it. I'm not sure how compatible this concept of just "providing an arbitrary private key via env var" is with the core concept of the `trin_data_dir`.

With this implementation, if somebody provides a private key via an env var. Then the node is started with a different identity than the data file to which the data is being written. 

Honestly, I'm not even sure if this is a feature we should support. 
- i'd prefer to minimize the # of conflicting config options for trin (env vars vs. cli args)
- there's been talk of moving all config into a `trin_config.yml` file, which I think would be nice

If this feature is something that we want to support, then it seems to me like we'll need to create a new `trin_data_dir` based off of the env var private key, and then write all of the data to the newly created data dir. this just seems a bit overkill imo.

Personally, I would hold off on implementing this until...
- we confirm that there's a significant ask for this feature
- we update to a `config.yml` file